core/src/main/java/com/bumble/appyx/core/node/ParentNode.kt

```diff
@@ -212,6 +221,48 @@ abstract class ParentNode<Routing : Any>(
         }
     }
 
+    protected suspend inline fun <reified T> attachWorkflow(
+        crossinline action: () -> Unit
+    ): T = withContext(lifecycleScope.coroutineContext + Dispatchers.Main) {
+        action()
+        val childNodes = children.value.values.mapNotNull { it.nodeOrNull }
+        val childNodesOfExpectedType = childNodes.filterIsInstance<T>()
+        if (childNodesOfExpectedType.isEmpty()) {
+            throw IllegalStateException(
+                "Expected child of type [${T::class.java}] was not found after executing action. " +
+                    "Check that your action actually results in the expected child. " +
+                    "Child count: ${childNodes.size}. " +
+                    "Last child is: [${childNodes.lastOrNull()}]. " +
+                    "All children: $childNodes"
+            )
+        } else {
+            childNodesOfExpectedType.last()
+        }
+    }
+
+    protected suspend inline fun <reified T : Node> waitForChildAttached(): T =
```

Comment:
Should we have an optional timeout in case the child is never attached?

Reply:
I think we shouldn't. It's fine for the job to hang around as long as needed.

Reply:
I like the idea of an optional timeout, as executing the original workflow after say half an hour could be unwanted and outdated. It's fine as a separate PR too.

Reply:
> I like the idea of an optional timeout, as executing the original workflow after say half an hour could be unwanted and outdated. It's fine as a separate PR too.

Why don't let clients have full control over workflows? If they need time-out they can implement it with a one-liner?

Reply:
Perhaps we should add some documentation to the method to make it clear that it will potentially block forever and that they should add their own timeout?

A lot of Java APIs that have await style methods usually have a timeout overload that makes it fairly obvious that not passing in a timeout value would cause it to wait forever

Reply:
Added a javadoc